src/test/unit_tests/test_system_store.js

```diff
@@ -138,4 +138,37 @@ mocha.describe('system_store', function() {
         assert.strictEqual(data2.systems[0].name, 'new_name');
     });
 
+    mocha.it('Check make_changes could not insert same id twice', async function() {
+        const system_id = system_store.new_system_store_id();
+        const orig_name = `JenTheMajesticSlothSystemStoreLoop3`;
+        await system_store.load();
+
+        await system_store.make_changes({
+            insert: {
+                systems: [{
+                    _id: system_id,
+                    name: orig_name,
+                    owner: system_store.new_system_store_id(),
+                }]
+            }
+        });
+        try {
+            await system_store.make_changes({
+                insert: {
+                    systems: [{
```

Comment:
For the second `make_changes` you should add a change\insert to different tables (pools, accounts, etc.). After the failure, make sure all changes were not applied. 
of course, this is the case for Postgres only, so we should skip this step for Mongo